indico/web/client/js/custom_elements/ind_time_picker.js

```diff
@@ -63,7 +71,24 @@ customElements.define(
 
       // Event handlers
 
+      indTimePicker.addEventListener('x-attrchange.value', () => {
+        const newValue = this.getAttribute('value');
+        CustomElementBase.setValue(input, newValue);
+
+        // Update the selection highlight
+        deselectCurrentSelection();
+        for (const option of listbox.children) {
+          if (option.dataset.value === newValue) {
+            markSelection(option);
+            break;
+          }
+        }
+      });
+
       input.addEventListener('click', () => {
+        // Select all text when clicking on the input
+        selectInputText(0);
```

Comment:
Doing it on focus would have nicer behavior (you can click again to select just parts of it / place the caret).